Tests/IntegrationTests/PSQLIntegrationTests.swift

```diff
@@ -360,4 +360,28 @@ final class IntegrationTests: XCTestCase {
         }
     }
     
+    func testConnectionClosureMidQueryDoesNotHang() async throws {
```

Comment:
Not sure if you're fine with me using async/await here. I could not quickly reproduce this with ELFs (the postgres connection would go out of scope and crash on deinit) but i think it should be possible.